xknx/cemi/cemi_frame.py

```diff
@@ -13,150 +13,176 @@
 """
 from __future__ import annotations
 
-from xknx.exceptions import ConversionError, UnsupportedCEMIMessage
+from abc import ABC, abstractmethod
+
+from xknx.exceptions import ConversionError, CouldNotParseCEMI, UnsupportedCEMIMessage
+from xknx.profile.const import ResourceObjectType, ResourcePropertyId
 from xknx.telegram import GroupAddress, IndividualAddress, Telegram
 from xknx.telegram.apci import APCI
 from xknx.telegram.tpci import TPCI, TDataBroadcast
 
-from .const import CEMIFlags, CEMIMessageCode
+from .const import CEMIErrorCode, CEMIFlags, CEMIMessageCode
 
 
-class CEMIFrame:
-    """Representation of a CEMI Frame."""
+class CEMIInfo:
+    """
+    Raw helper class for all CEMI additional info.
+
+    As specified within KNX Chapter 3.6.3/4.1.4.3 "Additional information".
+    """
+
+    def __init__(self, raw: bytes = bytes(0)):
+        """Initialize CEMILInfo object."""
+        self.raw = raw
+
+    def calculated_length(self) -> int:
+        """Get length of CEMI info."""
+        return 1 + len(self.raw)
+
+    @staticmethod
+    def from_knx(raw: bytes) -> tuple[CEMIInfo, bytes]:
+        """Parse/deserialize from CEMI raw data."""
+        length = raw[0]
+        return CEMIInfo(raw[1 : length + 1]), raw[length + 1 :]
+
+    def to_knx(self) -> bytes:
+        """Serialize to KNX/IP raw data."""
+        return bytes((len(self.raw),)) + self.raw
+
+    def __repr__(self) -> str:
+        """Return object as readable string."""
+        return f'CEMIInfo("{self.raw.hex()}")'
+
+    def __eq__(self, other: object) -> bool:
+        """Equal operator."""
+        return self.__dict__ == other.__dict__
+
+
+class CEMIData(ABC):
+    """Base class for all CEMI data."""
+
+    @abstractmethod
+    def calculated_length(self) -> int:
+        """Get length of CEMI data."""
+
+    @abstractmethod
+    def to_knx(self) -> bytes:
+        """Serialize to CEMI raw data."""
+
+    @classmethod
+    @abstractmethod
+    def from_knx(cls, raw: bytes) -> CEMIData:
+        """Parse/deserialize from KNX/IP raw data."""
+
+    @abstractmethod
+    def __repr__(self) -> str:
+        """Return object as readable string."""
+
+    def __eq__(self, other: object) -> bool:
+        """Equal operator."""
+        return self.__dict__ == other.__dict__
```

Comment:
```suggestion
    def __eq__(self, other: object) -> bool:
        """Equal operator."""
        if self.__class__ != other.__class__:
            return False
        return self.__dict__ == other.__dict__
```
Otherwise eg. `CEMIMPropReadResponse` would equal `CEMIMPropWriteRequest` with same data and property_info.